effekt/jvm/src/main/scala/effekt/Runner.scala

```diff
@@ -42,15 +42,44 @@ trait Runner[Executable] {
   /**
    * Creates a OS-specific script file that will execute the command when executed,
    * forwarding command line arguments.
+   * `$SCRIPT_DIR` refers to the directory the script is in.
+   * 
    * @return the actual name of the generated script (might be `!= name`)
    */
   def createScript(name: String, command: String*): String = os match {
     case OS.POSIX =>
-      IO.createFile(name, s"#!/bin/sh\n${command.mkString(" ")} \"$$@\"", true)
+      val computeScriptDir =
+        """# Determine the directory of the script
+          |if [ -n "$BASH_SOURCE" ]; then
+          |    # Bash
+          |    SCRIPT_DIR=$(dirname "$BASH_SOURCE")
+          |elif [ -n "$ZSH_VERSION" ]; then
+          |    # Zsh
+          |    SCRIPT_DIR=$(dirname "$0")
+          |elif [ -n "$KSH_VERSION" ]; then
+          |    # Ksh
+          |    SCRIPT_DIR=$(dirname "${.sh.file}")
+          |else
+          |    # POSIX shells (sh, dash, etc.)
+          |    SCRIPT_DIR=$(dirname "$0")
+          |fi
+          |""".stripMargin
```

Comment:
We could just try removing it and running the tests -- I don't think we really care about `ksh` support :)
```suggestion
      val computeScriptDir =
        """# Determine the directory of the script
          |SCRIPT_DIR=$(dirname "$0")
          |""".stripMargin
```